bindings/go/scip/symbol_formatter.go

```diff
@@ -44,6 +50,44 @@ var DescriptorOnlyFormatter = SymbolFormatter{
 	IncludePackageName:    func(_ string) bool { return false },
 	IncludePackageVersion: func(_ string) bool { return false },
 	IncludeDescriptor:     func(_ string) bool { return true },
+	IncludeRawDescriptor:  func(_ *Descriptor) bool { return true },
+	IncludeDisambiguator:  func(_ string) bool { return true },
+}
+
+// ReducedDescriptorOnlyFormatter formats a reduced descriptor omitting suffixes outside
+// of an explicit allowlist (currently including namespace, type, term, and method) as
+// well as method disambiguators.
+//
+// This formatter is meant to produce a "good enough" representation of the symbol that
+// can used to search for or match against a list of compiler-accurate SCIP symbols. The
+// suffixes in the allowlist are chosen as they are, in most cases, producible given only
+// a syntax tree.
+var ReducedDescriptorOnlyFormatter = SymbolFormatter{
+	OnError:               func(err error) error { return err },
+	IncludeScheme:         func(scheme string) bool { return scheme == "local" },
+	IncludePackageManager: func(_ string) bool { return false },
+	IncludePackageName:    func(_ string) bool { return false },
+	IncludePackageVersion: func(_ string) bool { return false },
+	IncludeDescriptor:     func(_ string) bool { return true },
+	IncludeRawDescriptor:  includeReducedRawDescriptor,
+	IncludeDisambiguator:  func(_ string) bool { return false },
+}
+
+var reducedSuffixes = []Descriptor_Suffix{
```

Comment:
Is it intentional to exclude parameters, type parameters, meta and macros? IIRC, the original goal was to only skip the method descriptor so that overloaded symbols would get the same symbol 🤔 The current "reduced" format seems a bit too ad-hoc to include in the official bindings.

Reply:
Meant method *disambiguator*

Reply:
Parameters and type parameters are very common suffixes. Is it intentional that we want the parameter of a method to be equal to the method itself?